egs/europarl_st/SRT/lcma_srt/decode.py

```
@@ -0,0 +1,1538 @@
+        else:
+            word_table = None
+            decoding_graph = k2.trivial_graph(params.vocab_size - 1, device=device)
+    else:
+        decoding_graph = None
+        word_table = None
+
+    if "modified_beam_search" in params.decoding_method:
+        if os.path.exists(params.context_file):
+            contexts = []
+            for line in open(params.context_file).readlines():
+                contexts.append((sp.encode(line.strip()), 0.0))
```

Ruff (0.15.20)

[error] Undefined name `sp`

(F821)